Introduction240

lindafoodev opened the pull request:
Added action, reducer, react-redux intro page component, edit app.js and store.js files to include introduction component/router and introReducer, respectively

Takes into consideration condition where user:
already viewed intro => setup to redirect (as discussed, not working pending you/scott futureconnection)
clicks after viewing final card => setup to redirect (as discussed, not working pending you/scott future connection)
views first intro card on mount component => shows appropriate first card
click for next card => shows appropriate next card

lindafoodev committed:
initial commit for intro component 240 - added intro component file

lindafoodev committed:
Merge branch 'master' of https://github.com/thinkful-ei17/teamThree-client into introduction240

lindafoodev committed:
comment out sections of app to get local development of intro started

lindafoodev committed:
cmt out requireslogin/user state in order to work local for intro

lindafoodev committed:
renamed intro for consistency, update app hard coded initial card data,

lindafoodev committed:
simplified introduction - working through iteration of cards

lindafoodev committed:
added number prop to introcard for easier ref, working to redirect

lindafoodev committed:
FIRST VERSION - REACT ONLY - INTRO PAGE WORKING (does not redirect)

lindafoodev committed:
refactor for redux - edit store/comp, add actions/reducer - rndr nxt

lindafoodev committed:
render and working - need to redirect

lindafoodev committed:
redirect not working-however intro comp is working as intended

lindafoodev committed:
refactor - only need one action for functionality

lindafoodev committed:
SECOND VERSION - REACT-REDUX INTRO PAGE WORKING (does not redirect)

lindafoodev committed:
revert cmts code to work local in order to PR

lindafoodev committed:
updated intro page to include introComplete

lindafoodev committed:
commit this version if local test works

lindafoodev committed:
added redirect if user already viewed intro - tested and working locally

lindafoodev committed:
removed test comments and enable requireslogin to intro comp

vampaynani commented on src/reducers/intro-reducer.js:
```diff
@@ -0,0 +1,39 @@
+    {number: 2, title: 'Bonds', details: 'Bonds Lorem Ipsum Concise'},
+    {number: 3, title: 'Managed Fund', details: 'Managed Fund Lorem Ipsum Concise'}],
+  numCard: 0,
+  introComplete: true
```
Intro complete needs to be `true` at the beginning?

lindafoodev replied:
Good catch - needs to default false at beginning, i had changed it to test the user condition - user already completed intro. Thanks!

lindafoodev replied:
Made changes to file

vampaynani reviewed (commented):
Store has conflicts with its master version, fix it before merge.
Move the return <Redirect .../> to the render method instead of being part of componentDidMount and nextCardClick. 
```
render(){
if(this.props.numCard === 3 || this.props.introComplete) return <Redirect to="/dashboard" />
...
}
```

lindafoodev committed:
updated intro reducer based on review comment

changed intoCompleted default from true to false. False = user has not completed intro.

ASTribble committed:
Merge branch 'master' into introduction240

ASTribble commented:
The conflicts in the store were with making sure we wanted to add  another new reducer.  
Linda and I looked at them together, didn't commit the changes, so I re-implemented those changes.

lindafoodev committed:
Move redirect to render 

Move the return <Redirect.../> to the render method instead of being part of componentDidMount and nextCardClick.

lindafoodev committed:
Merge branch 'master' into introduction240
